### PR TITLE
Feat:  support label selector filter for 'vela ls'

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

### Description of your changes
support label selector filter for 'vela ls'

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by wonderflow — COMMENTED** [review]
let's add some test case

**Comment by nuclearwu** [comment]
> let's add some test case

Done

**Comment by wonderflow** [comment]
```

--- FAIL: TestBuildApplicationListTable (0.01s)
    --- FAIL: TestBuildApplicationListTable/specified_label_selector (0.00s)
        ls_test.go:113: 
            	Error Trace:	ls_test.go:113
            	Error:      	Received unexpected error:
            	            	applications.core.oam.dev "dependency" already exists
            	Test:       	TestBuildApplicationListTable/specified_label_selector
```

**Comment by nuclearwu** [comment]
> ```
> 
> --- FAIL: TestBuildApplicationListTable (0.01s)
>     --- FAIL: TestBuildApplicationListTable/specified_label_selector (0.00s)
>         ls_test.go:113: 
>             	Error Trace:	ls_test.go:113
>             	Error:      	Received unexpected error:
>             	            	applications.core.oam.dev "dependency" already exists
>             	Test:       	TestBuildApplicationListTable/specified_label_selector
> ```
ok, done

**Comment by wonderflow** [comment]
CI is still failing

**Comment by nuclearwu** [comment]
> CI is still failing

ok, done

**Review by wonderflow — APPROVED** [review]

File: references/cli/ls_test.go [hunk 1]
```diff
@@ -83,6 +84,22 @@ func TestBuildApplicationListTable(t *testing.T) {
...
+					Labels: map[string]string{
+						"app.kubernetes.io/name":    "nginx",
+						"app.kubernetes.io/version": "v1",
+					},
+				},
+				Spec:   componentOrderSpec,
+				Status: componentOrderStatus,
+			},
+			expectedErr:   nil,
+			labelSelector: "app.kubernetes.io/name=nginx,app.kubernetes.io/version=v1",
```
Comment (chivalryq) [inline references/cli/ls_test.go:101]: I remove this line and find the test passes successfully.

**Review by chivalryq — CHANGES_REQUESTED** [review]
Feature is not actually tested. You can always list app in `test` namespace. You should change the `tc.app` to `tc.apps` list and test it list only apps with labels.

File: references/cli/ls_test.go (hunk 1 above)
Comment (nuclearwu) [reply references/cli/ls_test.go:101]: > I remove this line and find the test passes successfully.

Because the labelSelector is optional, not mandatory

**Comment by nuclearwu** [comment]
> Feature is not actually tested. You can always list app in `test` namespace. You should change the `tc.app` to `tc.apps` list and test it list only apps with labels.

ok, done

File: references/cli/ls_test.go [hunk 2]
```diff
@@ -91,26 +144,38 @@ func TestBuildApplicationListTable(t *testing.T) {
...
+					r.Equal(string(app.Status.Phase), fmt.Sprintf("%s", row.Cells[4].Data))
+					r.Equal(getHealthString(service[compName].Healthy), fmt.Sprintf("%s", row.Cells[5].Data))
+					r.Equal(service[compName].Message, fmt.Sprintf("%s", row.Cells[6].Data))
+				}
+			}
+			if len(tc.labelSelector) > 0 {
+				appName := fmt.Sprintf("%s", tb.Rows[1].Cells[0].Data)
+				r.Equal(len(tc.apps)-1, len(tb.Rows)-1)
+				r.Equal(tc.apps[1].Name, appName)
 			}
```
Comment (chivalryq) [inline references/cli/ls_test.go:179]: This snippet works only in the case:
1. test case have x app and (x-1) app have label.
2. second app of test case is labeled and listed.

Please consider making the test more robust, for example add `expectedAppList int` field 
 to testCases.

Comment (nuclearwu) [reply references/cli/ls_test.go]: Done

**Review by chivalryq — APPROVED** [review]
LGTM

**Pull request merged**